requester/report.go

```diff
@@ -21,38 +20,32 @@ func (rep Report) String() string {
 	return string(b)
 }
 
-// report generates and returns a Report from a previous Run.
-func makeReport(records []Record, numErr int) Report {
-	return Report{
-		Records: records,
-		Length:  len(records),
-		Success: len(records) - numErr,
-		Fail:    numErr,
+// Stats returns basic stats about the report's records:
+// min duration, max duration, and mean duration.
+// It does not replace the remote computing and should only be used
+// when a local reporting is needed.
+func (rep Report) Stats() (min, max, mean time.Duration) {
+	var sum time.Duration
+	for _, rec := range rep.Records {
+		d := rec.Time
+		if d < min || min == 0 {
+			min = d
+		}
+		if d > max {
+			max = d
+		}
+		sum += rec.Time
 	}
+	return min, max, sum / time.Duration(rep.Length)
 }
 
-// SendReport sends the report to url. Returns any non-nil error that occurred.;
-//
-// TODO: move from requester
-func SendReport(url string, report Report) error {
-	body := bytes.Buffer{}
-	if err := json.NewEncoder(&body).Encode(report); err != nil {
-		return fmt.Errorf("%w: %s", ErrReporting, err)
-	}
-
-	req, err := http.NewRequest("POST", url, &body)
-	if err != nil {
-		return fmt.Errorf("%w: %s", ErrReporting, err)
-	}
-
-	resp, err := http.DefaultClient.Do(req)
-	if err != nil {
-		return fmt.Errorf("%w: %s", ErrReporting, err)
-	}
-	defer resp.Body.Close()
-	if resp.StatusCode >= 400 {
-		return fmt.Errorf("%w: %s", ErrReporting, resp.Status)
+// makeReport generates and returns a Report from a previous Run.
+func makeReport(records []Record, numErr int, d time.Duration) Report {
```

Comment:
As it only instanciate a new report without doing anything else: 
```suggestion
func newReport(records []Record, numErr int, d time.Duration) Report {
```
?